### PR TITLE
Fix: Bug in no-extra-bind (fixes #4806)

### LINKED ISSUE #4806: Bug in no-extra-bind
There is a bug in the symmetry of pushing/popping on the `scope` stack in the code of this rule.
The `pop` is skipped in case of a necessary `bind` which breaks cases like this one:

```javascript
var a = function() {
  (function(){
    (function(){ this.d }.bind(c))
  })
}.bind(b)
```

`bind(b)` is unnecessary, but the error does not get reported.

I already have a fix and created the issue only so I could submit the pull request.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29401527-bug-in-no-extra-bind?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mjomble**
Fix for a scoping bug in no-extra-bind

### CODE REVIEW

**Comment by ilyavolodin** [comment]
LGTM

File: tests/lib/rules/no-extra-bind.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ ruleTester.run("no-extra-bind", rule, {
         { code: "var a = (() => { return 1; }).bind(b)", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "The function binding is unnecessary.", type: "CallExpression"}] },
         { code: "var a = (() => { return this; }).bind(b)", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "The function binding is unnecessary.", type: "CallExpression"}] },
         { code: "var a = function() { (function(){ this.c }) }.bind(b)", errors: [{ message: "The function binding is unnecessary.", type: "CallExpression"}] },
-        { code: "var a = function() { function c(){ this.d } }.bind(b)", errors: [{ message: "The function binding is unnecessary.", type: "CallExpression"}] }
+        { code: "var a = function() { function c(){ this.d } }.bind(b)", errors: [{ message: "The function binding is unnecessary.", type: "CallExpression"}] },
+        { code: "var a = function() { (function(){ (function(){ this.d }.bind(c)) }) }.bind(b)", errors: [{ message: "The function binding is unnecessary.", type: "CallExpression"}] }
```
Comment (btmills) [inline tests/lib/rules/no-extra-bind.js]: Can you add a column to the message to verify for any future changes that it still warns on `bind(b)` and not `bind(c)`?

Comment (mjomble) [inline tests/lib/rules/no-extra-bind.js]: Do you mean adding the info to the message string?
Or can I specify the column as a property of the expected error object?
The guide at http://eslint.org/docs/developer-guide/working-with-rules only has examples with `message` and `type`.

Comment (btmills) [inline tests/lib/rules/no-extra-bind.js]: [Here's an example](https://github.com/eslint/eslint/blob/3642d0bb6ced17eeed50c030531a3ddbeb939f29/tests/lib/rules/no-alert.js#L43).

I created #4810 to add those properties to the docs.

**Comment by mjomble** [comment]
Added the column parameter

**Pull request merged**